### PR TITLE
Fetch only requested

### PR DESCRIPTION
**Pull request by vogel**

### CODE REVIEW

File: src/overlay/ItemFetcher.h [hunk 1]
```diff
@@ -38,87 +25,144 @@ class TxSetFrame;
...
-    using TrackerPtr = std::shared_ptr<TrackerT>;
-
-    explicit ItemFetcher(Application& app);
-
-    void fetch(uint256 itemID, const SCPEnvelope& envelope);
-
+    using TrackerPtr = std::shared_ptr<Tracker>;
+
+    /**
+     * Create ItemFetcher that fetches data using @p askPeer delgate.
```
Comment (raymens) [inline src/overlay/ItemFetcher.h]: *delegate* typo

File: src/overlay/ItemFetcher.h [hunk 2]
```diff
@@ -14,19 +14,6 @@
 #include <util/optional.h>
 #include "util/HashOfHash.h"
 
-/*
```
Comment (jedmccaleb) [inline src/overlay/ItemFetcher.h]: Why did you take this comment out?

**Comment by jedmccaleb** [comment]
r+ except for the one comment

File: src/overlay/ItemFetcher.h (hunk 2 above)
Comment (vogel) [inline src/overlay/ItemFetcher.h]: This comment was mostly made invalid in this commit: https://github.com/stellar/stellar-core/commit/7a86fff5ed8ad009faed7856c7e9b5e411c305ef:
* ItemFetcher does not return instances of the Tracker class
* there are no longed shared_ptrs of Tracker instances

Remaining valid sentences were moved directly before ItemFetcher class:

+/**
 + * @class ItemFetcher
 + *
 + * Manages asking for Transaction or Quorum sets from Peers
 + *
 + * The ItemFetcher keeps instances of the Tracker class. There exists exactly
 + * one Tracker per item. The tracker is used to maintain the state of the
 + * search.
 + */

**Pull request merged**